pkg/test/nodeclaim.go

```diff
@@ -57,7 +65,9 @@ func NodeClaim(overrides ...v1.NodeClaim) *v1.NodeClaim {
 
 func NodeClaimAndNode(overrides ...v1.NodeClaim) (*v1.NodeClaim, *corev1.Node) {
 	nc := NodeClaim(overrides...)
-	return nc, NodeClaimLinkedNode(nc)
+	node := NodeClaimLinkedNode(nc)
+	nc.Status.NodeName = node.Name
```

Comment:
This changes the logic, right? This now schedules the node when it wasn't scheduled before?

Reply:
Not really, we don't rely on this anywhere for testing (in this package). In my mind, `NodeClaimAndNode` should return a pair which have already been linked together, reflecting that in the status is appropriate.

Reply:
I can drop for now then, this was helping out some tests that assume this link exists but I can just add it there. Longer term I think we should include some test helpers that give us a Node and NodeClaim at each stage of the lifecycle (e.g. registered, initialized, terminating).